src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -510,15 +519,27 @@ public void scan() {
 					}
 				}
 			}
+			super.scan();
 		} finally {
 			this.returnFirefoxDriver(fxDriver);
 		}
 	}
 
-	public boolean scan(ArrayList<WebDriverWrapper> drivers, String attackVector) {
-    	HttpMessage msg = getBaseMsg();
-    	String url = msg.getRequestHeader().getURI().toString();
-    	String currURL = url + attackVector;
+	public boolean scan(ArrayList<WebDriverWrapper> drivers, String paramName, String attackVector, String uri) {
+		String currURL;
+		HttpMessage msg = getNewMsg();
+		if (!uri.isEmpty()) {
+        	currURL = uri + attackVector;
+		} else {
+			TreeSet<HtmlParameter> urlParams = msg.getUrlParams();
+			for (HtmlParameter param : urlParams) {
+				if (param.getName().equals(paramName)) {
+					param.setValue(attackVector);
+				}
+			}
+			msg.setGetParams(urlParams); // Payloads aren't escaped hence not using setEscapedParameter
```

Comment:
Why not use `setParameter` then? :) I was expecting a mention that the variant does not escape correctly for this use case (`+` instead of `%20`).

Reply:
I dunno....because they're GET params.....

Reply:
The variant being used if for GET parameters too.

Reply:
I honestly hadn’t thought about it. I just made it work. I hadn’t been thinking “how do other appparam plugins work?”.....though I probably should have.

Reply:
I’ve updated the comment.

Reply:
Thanks.